`Error displaying widget` for ImageTexture in Textures example with JupyterLab

The two ImageTexture objects `checker_tex` and `earth_tex` cannot be rendered with jupyterlab.
`data_tex` can be rendered correctly.

With a jupyter notebook everything works as expected.

The console gives the following output:
```
future.js:155 Uncaught (in promise) Error: Canceled
    at t.dispose (future.js:155)
    at default.js:385
    at Map.forEach (<anonymous>)
    at e.dispose (default.js:384)
    at e.dispose (default.js:244)
    at index.js:241
    at <anonymous>
checkerboard.png Failed to load resource: the server responded with a status of 404 (Not Found)
ImageTexture.js:22 Error loading texture: img/checkerboard.png
bluebird.js:1545 Warning: a promise was rejected with a non-error: [object Event]
    at U (http://localhost:8889/lab/static/main.75a01e1….js:1:109001)
    at A.t._warn (http://localhost:8889/lab/static/main.75a01e1….js:1:103042)
    at A._rejectCallback (http://localhost:8889/lab/static/main.75a01e1….js:1:144410)
    at http://localhost:8889/lab/static/main.75a01e1….js:1:144677
    at http://localhost:8889/lab/static/main.75a01e1….js:1:2017011
    at HTMLImageElement.<anonymous> (http://localhost:8889/lab/static/main.75a01e1….js:1:1368742)
From previous event:
    at A.R [as _captureStackTrace] (http://localhost:8889/lab/static/main.75a01e1….js:1:107952)
    at A._resolveFromExecutor (http://localhost:8889/lab/static/main.75a01e1….js:1:144554)
    at new A (http://localhost:8889/lab/static/main.75a01e1….js:1:137678)
    at i.constructThreeObjectAsync (http://localhost:8889/lab/static/main.75a01e1….js:1:2016837)
    at i.createThreeObjectAsync (http://localhost:8889/lab/static/main.75a01e1….js:1:949721)
    at initialize (http://localhost:8889/lab/static/main.75a01e1….js:1:947950)
    at i.t.Model (http://localhost:8889/lab/static/main.75a01e1….js:1:1900366)
    at i.t [as constructor] (http://localhost:8889/lab/static/main.75a01e1….js:1:2212313)
    at i [as constructor] (http://localhost:8889/lab/static/main.75a01e1….js:1:1918529)
    at i [as constructor] (http://localhost:8889/lab/static/main.75a01e1….js:1:1918529)
    at i [as constructor] (http://localhost:8889/lab/static/main.75a01e1….js:1:1918529)
    at new i (http://localhost:8889/lab/static/main.75a01e1….js:1:1918529)
    at t.<anonymous> (http://localhost:8889/lab/static/main.75a01e1….js:1:2491868)
    at l (http://localhost:8889/lab/static/main.75a01e1….js:1:2487926)
    at Object.next (http://localhost:8889/lab/static/main.75a01e1….js:1:2487257)
    at a (http://localhost:8889/lab/static/main.75a01e1….js:1:2486784)
    at <anonymous>
```

--- a/js/package.json
+++ b/js/package.json
@@ -31,7 +31,7 @@
     "webpack-cli": "^2.0.10"
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^1.1.9",
+    "@jupyter-widgets/base": "^1.1.10",
     "bluebird": "^3.5.1",
     "jupyter-dataserializers": "^1.1.2",
     "three": "^0.90.0",

--- a/js/package.json
+++ b/js/package.json
@@ -31,7 +31,7 @@
     "webpack-cli": "^2.0.10"
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^1.1.8",
+    "@jupyter-widgets/base": "^1.1.9",
     "bluebird": "^3.5.1",
     "jupyter-dataserializers": "^1.1.2",
     "three": "^0.90.0",

--- a/js/src/textures/ImageTexture.js
+++ b/js/src/textures/ImageTexture.js
@@ -7,22 +7,25 @@ var ImageTextureModel = ImageTextureBase.ImageTextureModel.extend({
     constructThreeObjectAsync: function() {
 
         var loader = new THREE.TextureLoader();
-        var imageUri = this.get('imageUri');
+        // Ensure we resolve any local paths according to current notebook location:
+        var imageUriPromise = this.widget_manager.resolveUrl(this.get('imageUri'));
 
         var p = new Promise(function(resolve, reject) {
-            loader.load(
-                imageUri,
-                function(texture) {
-                    return resolve(texture);
-                },
-                function(xhr) {
-                    console.debug(imageUri + ': ' + (xhr.loaded / xhr.total * 100) + '%');
-                },
-                function(xhr) {
-                    console.log('Error loading texture: ' + imageUri);
-                    return reject(xhr);
-                }
-            );
+            imageUriPromise.then(function (imageUri) {
+                loader.load(
+                    imageUri,
+                    function(texture) {
+                        return resolve(texture);
+                    },
+                    function(xhr) {
+                        console.debug(imageUri + ': ' + (xhr.loaded / xhr.total * 100) + '%');
+                    },
+                    function(xhr) {
+                        console.log('Error loading texture: ' + imageUri);
+                        return reject(xhr);
+                    }
+                );
+            }, reject);
         });
         return p;
     },

--- a/setup.py
+++ b/setup.py
@@ -68,7 +68,7 @@ setup_args = {
     'license': 'BSD',
     'include_package_data': True,
     'install_requires': [
-        'ipywidgets>=7.2',
+        'ipywidgets>=7.2.1',
         'ipydatawidgets>=1.1.1',
         'numpy',
     ],

--- a/setup.py
+++ b/setup.py
@@ -68,7 +68,7 @@ setup_args = {
     'license': 'BSD',
     'include_package_data': True,
     'install_requires': [
-        'ipywidgets>=7,<8',
+        'ipywidgets>=7.2',
         'ipydatawidgets>=1.1.1',
         'numpy',
     ],

